libs/wordpress-plugin/plugin/trunk/server/query.php

```diff
@@ -71,42 +71,59 @@ function block_protocol_extract_filter_value($filter, $raw = false)
   return block_protocol_encode_filter_value($filter_value);
 }
 
-function block_protocol_build_string_filter($filter_value, $json_path, $operator, &$sql_fragment, &$values)
+/**
+ * This function will return a SQL string that casts text to JSON. 
+ * Note that this is function is "unsafe" and shouldn't have any direct user data passed into it.
+ */
+function block_protocol_db_json_unsafe_cast(string $value, bool $is_maria_db)
+{
+  if($is_maria_db) {
+    return "JSON_COMPACT({$value})";
```

Comment:
From what I can tell `JSON_COMPACT` even has a binary/non-string representation in the DB in version 10.2.7, so maybe the checks are more efficient when we do this casting? (although there's overhead in the calls..)